### PR TITLE
Apply linter changes 

### PR DESCRIPTION
**Pull request by oliviamiller**

### CODE REVIEW

File: gateway/sensor.go [hunk 1]
```diff
@@ -188,16 +191,15 @@ func (g *Gateway) receivePackets() {
...
-					for i := 0; i < int(packet.size); i++ {
-						payload = append(payload, byte(packet.payload[i]))
-					}
-					g.handlePacket(ctx, payload)
+				if packet.size == 0 {
+					continue
 				}
+				// Convert packet to go byte array
+				//nolint:intrange
+				for i := 0; i < int(packet.size); i++ {
```
Comment (randhid) [inline gateway/sensor.go:199]: you can do i := range packet.size in go 1.23 i believe

File: gateway/uplink.go [hunk 2]
```diff
@@ -181,7 +183,7 @@ func executeDecoder(ctx context.Context, script string, vars map[string]interfac
 	}
 
 	resultChan := make(chan result)
-	timeoutCtx, _ := context.WithTimeout(ctx, 10*time.Millisecond)
+	timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Millisecond)
```
Comment (randhid) [inline gateway/uplink.go:186]: let's make sure this new behaviour works as expected.

**Review by randhid — APPROVED** [review]
manually test the behaviour changes and good to go

File: gateway/sensor.go [hunk 3]
```diff
@@ -188,16 +191,15 @@ func (g *Gateway) receivePackets() {
...
-					// Convert packet to go byte array
-					for i := 0; i < int(packet.size); i++ {
-						payload = append(payload, byte(packet.payload[i]))
-					}
-					g.handlePacket(ctx, payload)
+				if packet.size == 0 {
+					continue
 				}
+				// Convert packet to go byte array
+				//nolint:intrange
```
Comment (martha-johnston) [inline gateway/sensor.go:198]: you can also add this to the `disable` section of your `etc/.golangci.yaml` file

File: gateway/uplink.go [hunk 4]
```diff
@@ -119,6 +118,7 @@ func matchDeviceAddr(devAddr []byte, devices map[string]*node.Node) (*node.Node,
 }
 
 func decodePayload(ctx context.Context, fPort uint8, path string, data []byte) (map[string]interface{}, error) {
+	//nolint:gosec
```
Comment (martha-johnston) [inline gateway/uplink.go:121]: same here, can add to `disable` section, but it's up to your discretion if you think this case is an exception, or if you'd always want to disable this. when I was modularizing the builtins, if I ran into lint issues like this I would check if it was disabled in the rdk `golangci.yaml` file, and if so I would disable it in the module as well

File: gateway/uplink.go [hunk 5]
```diff
@@ -192,15 +194,15 @@ func executeDecoder(ctx context.Context, script string, vars map[string]interfac
 	select {
 	case <-timeoutCtx.Done():
 		vm.Interrupt <- func() {
-			errors.New("ctx canceled")
 		}
+		cancel()
```
Comment (martha-johnston) [inline gateway/uplink.go:198]: maybe can `defer cancel()` above this instead? unless there's a situation in which you wouldn't want to call cancel before this function returns

**Review by martha-johnston — APPROVED** [review]

File: gateway/sensor.go (hunk 1 above)
Comment (oliviamiller) [reply gateway/sensor.go:199]: I tried that and it errored

File: gateway/uplink.go (hunk 4 above)
Comment (oliviamiller) [reply gateway/uplink.go:121]: I will disable this, thanks!

File: gateway/sensor.go (hunk 1 above)
Comment (randhid) [reply gateway/sensor.go:199]: when cast to an int as well?

Comment (oliviamiller) [reply gateway/sensor.go]: tried it again visual studio shows it as an error but it still builds and works so changed it!

File: gateway/uplink.go (hunk 2 above)
Comment (oliviamiller) [reply gateway/uplink.go:186]: tested that module still works correctly with all these changes!

File: etc/.golangci.yaml [hunk 6]
```diff
@@ -46,6 +46,7 @@ linters:
     - varnamelen
     - wrapcheck
     - wsl
+    - gosec
```
Comment (JohnN193) [inline etc/.golangci.yaml:49]: Don’t disable all of gosec, just the one or two codes that you have to have for the implementation. rdk+other modules with linters should have an example

File: gateway/sensor.go [hunk 7]
```diff
@@ -374,21 +377,26 @@ func convertToBytes(key interface{}) ([]byte, error) {
...
+		err := gpio.ResetGPIO(g.rstPin, *g.bookworm)
+		if err != nil {
+			g.logger.Error("error reseting gateway")
+		}
 	}
 
 	if g.workers != nil {
 		g.workers.Stop()
 	}
 	errCode := C.stopGateway()
```
Comment (JohnN193) [inline gateway/sensor.go:392]: [q]Would printing the error code be useful for debugging

**Review by JohnN193 — APPROVED** [review]
Just a heads up I’m off until next Tuesday, so I probably won’t be responsive on these PRs!

File: gateway/sensor.go (hunk 7 above)
Comment (oliviamiller) [reply gateway/sensor.go:392]: no its always 1

Comment (oliviamiller) [reply gateway/sensor.go:392]: I have a ticket to better handle and surface the errors from the C code though so may readdress this there

File: gateway/sensor.go (hunk 1 above)
Comment (randhid) [reply gateway/sensor.go]: ah! your vscode has older golang in its environment.

**Pull request merged**